internal/pkg/cli/deploy/static_site.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package deploy
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/partitions"
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/upload/asset"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/manifest/manifestinfo"
+	"github.com/spf13/afero"
+)
+
+type staticSiteDeployer struct {
+	*svcDeployer
+	staticSiteMft *manifest.StaticSite
+	bucketName    string
+	fs            afero.Fs
+	uploadFn      func(fs afero.Fs, source, destination string, opts *asset.UploadOpts) ([]string, error)
+}
+
+// NewStaticSiteDeployer is the constructor for staticSiteDeployer.
+func NewStaticSiteDeployer(in *WorkloadDeployerInput) (*staticSiteDeployer, error) {
+	svcDeployer, err := newSvcDeployer(in)
+	if err != nil {
+		return nil, err
+	}
+	mft, ok := in.Mft.(*manifest.StaticSite)
+	if !ok {
+		return nil, fmt.Errorf("manifest is not of type %s", manifestinfo.StaticSiteType)
+	}
+	return &staticSiteDeployer{
+		svcDeployer:   svcDeployer,
+		staticSiteMft: mft,
+		fs:            afero.NewOsFs(),
+		bucketName:    svcDeployer.resources.S3Bucket,
```

Comment:
Is `svcDeployer.resources` guaranteed to have been populated in `newSvcDeployer` ? Do we need some sort of comment in `newSvcDeployer` so that we remember to grab the resources in  `svcDeployer.resources` in `newSvcDeployer`?

Reply:
I think it's ok. Both `newSvcDeployer` and `newWorkloadDeployer` initialize every thing for the struct.